tests/MongoDB.Driver.Tests/MongoCollectionImplTests.cs

```diff
@@ -1541,7 +1547,8 @@ private class ClassForDistinctWithArrayField
                 TextIndexVersion = 60,
                 Unique = true,
                 Version = 70,
-                Weights = weights
+                Weights = weights,
+                WildcardProjection = wildcardProjection
```

Comment:
Since `WildcardProjection` is `BsonDocument` and `CreateIndexOptions` is not a generic class, currently we cannot use the projection builder for this property. 
But we can do something like this (in CreateIndexOptions<TDocument>):

    public new ProjectionDefinition<TDocument> WildcardProjection { get; set; }

it gives us ability to construct this property in this way:

    var indexOptions = new CreateIndexOptions<BsonDocument>
    {
         WildcardProjection = Builders<BsonDocument>.Projection.Include("a").Include("b")
    };

Just thoughts, it doesn't look like an ideal solution.

Reply:
Noted. I'll defer to @rstam as to how he'll want to handle this

Reply:
I have one more idea. I will add it tomorrow morning.

Reply:
As I see, the nongeneric `CreateIndexOptions` class is almost `obsolete`. The only place where this class is expected and where it's not `obsolete` is in `CreateIndexModel` constructor. But inside the constructor, we convert the nongeneric option into a generic. 
I guess we can add the `WildcardProjection` only into a generic version of `CreateIndexOption` and with `ProjectionDefinition` type like this:

>public ProjectionDefinition<TDocument> WildcardProjection

This property will be similar to `PartialFilterExpression`. @rstam what do you think?

Reply:
I think you should try to use something more safe than BsonDocument for WildcardProjection.

Reply:
Yep, but how about my above suggestion about having `WildcardProjection` similar to `PartialFilterExpression`? But in this case, we will not have a `WildProjection` property in nongeneric `CreateIndexOptions`?

Reply:
Done